plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/data/formatters/DateTimeDataFormatter.java

```diff
@@ -79,11 +81,22 @@ public DateFormat getDateFormat() {
     }
 
     @Override
-    public String getPattern()
-    {
+    public String getPattern() {
         return pattern;
     }
 
+    @Override
+    public void setPattern(@NotNull String pattern) {
+        this.pattern = pattern;
+        String sdfPattern = pattern.replace("n", "f");
+        dateFormat = new ExtendedDateFormat(sdfPattern, locale);
+        dateFormat.setLenient(false);
+        buffer = new StringBuffer();
+        position = new FieldPosition(0);
+        String java8DatePattern = pattern.replaceAll("f+", "n");
+        dateTimeFormatter = DateTimeFormatter.ofPattern(java8DatePattern);

```

Comment:
It looks like a copy paste from `init` method. Maybe it's better to use `init` instead of creating such a strange "setter".